modules/yarn.lock

```diff
@@ -10401,14 +10412,21 @@ metal-path-parser@1.0.4, metal-path-parser@^1.0.3:
   dependencies:
     metal "^2.16.6"
 
-metal-position@2.1.2, metal-position@^2.1.0, metal-position@^2.1.2:
+metal-position@2.1.2, metal-position@^2.0.0, metal-position@^2.1.0, metal-position@^2.1.2:
   version "2.1.2"
   resolved "https://registry.yarnpkg.com/metal-position/-/metal-position-2.1.2.tgz#3643a7f83a6b5a39061841858b058935390bd13c"
   integrity sha512-XvE1AQoySYFGG0sc3unZWfK97UKOvHvg/asq2k1Oh+vs1NGD9PhhGBQKiD0/Rix85ICHEeLUVzS0WGQr5wdapQ==
   dependencies:
     metal "^2.16.6"
     metal-dom "^2.16.6"
 
+metal-promise@3.0.5:
+  version "3.0.5"
+  resolved "https://registry.yarnpkg.com/metal-promise/-/metal-promise-3.0.5.tgz#03d8fb40579e5b97a4e44356e00042f2d34844de"
+  integrity sha512-9LRn5oPcv439xCiNvmKBWVopFBaJ5zBV75lkZMydufJX17YJ8J6+EVrL2ctlrxxxyF/hFhiEGVGPvxNjY9kwxg==
+  dependencies:
+    metal "^2.16.6"
+
```

Comment:
It would be best if we could avoid having two versions of `metal-promise` in the lockfile, but looking at [the commit where you removed it](https://github.com/liferay/liferay-portal/commit/5ce64fb9c9f8458539d9975e394da83a977fee76) I see that it was duplicated back then too... 😢

`yarn why metal-promise` says:

```
=> Found "metal-promise@2.0.1"
info Has been hoisted to "metal-promise"
info Reasons this module exists
   - "workspace-aggregator-1ef95de1-a159-40d8-b90e-42b40cb4ef92" depends on it
   - Hoisted from "_project_#frontend-js-metal-web#metal-ajax#metal-promise"
   - Hoisted from "_project_#frontend-js-metal-web#metal-router#metal-promise"
   - Hoisted from "_project_#frontend-js-metal-web#metal-router#senna#metal-promise"
info Disk size without dependencies: "88KB"
info Disk size with unique dependencies: "300KB"
info Disk size with transitive dependencies: "300KB"
info Number of shared dependencies: 1
=> Found "frontend-js-metal-web#metal-promise@3.0.5"
info This module exists because "_project_#frontend-js-metal-web" depends on it.
info Disk size without dependencies: "144KB"
info Disk size with unique dependencies: "356KB"
info Disk size with transitive dependencies: "356KB"
info Number of shared dependencies: 1
```

So we have our own metal dependencies pulling in an older version of a metal dependency... 🤦‍♂️ I guess we can forward this without making things any worse than they were originally, but there might be an opportunity here to clean up a bit 🤔

Looking at [the history of metal-promise](https://github.com/metal/metal-plugins/commits/master/packages/metal-promise), these are the changes from 2.0.1 to 3.0.5:

```
git diff -w --stat ac0bcb1e15ff8acdb2fee84f6b0e692fa0b59b19 282bfcb63137ce4b9982f91a7a10a7eb8f2ab15a -- packages/metal-promise
 packages/metal-promise/.babelrc                        |    5 +
 packages/metal-promise/LICENSE.md                      |   29 +
 packages/metal-promise/README.md                       |   96 +
 packages/metal-promise/__tests__/CancellablePromise.js |  510 +++++
 packages/metal-promise/__tests__/ProgressPromise.js    |  143 ++
 packages/metal-promise/package-lock.json               | 6235 +++++++++++++++++++++++++++++++++++++++++++++++++++++++
 packages/metal-promise/package.json                    |   35 +-
 packages/metal-promise/src/CancellablePromise.js       | 1441 +++++++++++++
 packages/metal-promise/src/FreeList.js                 |   86 +
 packages/metal-promise/src/ProgressPromise.js          |  150 ++
 packages/metal-promise/src/Thenable.js                 |  100 +
 packages/metal-promise/src/all.js                      |    5 +
 packages/metal-promise/src/promise/Promise.js          |  954 ---------
 13 files changed, 8820 insertions(+), 969 deletions(-)
```

So, yeah... looks like a boatload of changes under `src`; the contents of `src/promise/Promise.js` were factored out into separate files, and it looks like a lot was added at the same time. The [commit message that introduced the change](https://github.com/metal/metal-plugins/commit/c8d1c011dd09002f33da8cd635f3609ef07fa01f) doesn't explain what happened or why, and it is also mislabeled as a "Merge" commit (even though it only has one parent) and [the PR that it references](https://github.com/metal/metal-plugins/pull/19) is totally unrelated. I was kind of hoping that there would be no substantive difference between them and we could just "standardize" on 2.0.5, but seems like the answer is no.